[Ng Hong Jie Ryan] iP

nhjryan opened the pull request:
# Welcome to Duke!
> _"Victors are the sole authors of history. Time to write our chapter."_

_Duke_ is a Personal Assistant that tracks your:
1. To-dos :ballot_box_with_check:
2. Deadlines :watch:
3. Events :tada:

and allows you to:
* **Add** and ~~delete~~ tasks
* Mark your tasks as done
* Save and load your list of tasks

Features:
- [X] Save and Load
- [ ] Find
- [ ] GUI (coming soon)
````
if (interested) {
  click on the link below!
}
````
Click on [`this`](https://github.com/nhjryan) link!

_Change the way you manage your life with Duke!_

Commit:
Add Gradle support

nhjryan committed:
Added Level 1. Greet, Echo, Exit

nhjryan committed:
Added Level 2. Add, List + minor edits to Level 1

nhjryan committed:
Added Level 3. Mark as Done

nhjryan committed:
Added Level 4. ToDos, Events, Deadlines

nhjryan committed:
Added Automated Text UI Testing + fixed minor errors in Duke

nhjryan committed:
Added Level 6. Delete and necessary unit tests + minor edits in Duke

Made minor edits to indentation and replaced topLine and bottomLine string with a msgTemplate method.

nhjryan committed:
Add Level 5. Handle Errors and relevant test cases

nhjryan committed:
Import java.io.File

cedricongjh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,101 @@
+                break;
+            } else {
+                msgTemplate(" ☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
```
Great avoidance of deep nesting!

cedricongjh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,101 @@
+    public static void msgTemplate(String msg){
+        String topLine = "____________________________________________________________\n";
+        String bottomLine = "____________________________________________________________";
+        System.out.println(topLine + msg + System.lineSeparator() + bottomLine);
```
Good avoidance of repeating code by using a function to generate the message

cedricongjh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,101 @@
-        System.out.println("Hello from\n" + logo);
+        String startUp = " Hello! I'm Duke\n"
+                + " What can I do for you?";
+        ArrayList<Task> list = new ArrayList<>();
```
Perhaps a better name for this list could be taskList, or tasks, to make it clearer as to what the list contains

cedricongjh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,101 @@
+                }
+            } else if (input.startsWith("deadline")) {
+                try {
+                    if (input.charAt(8) != ' ') throw new IllegalArgumentException();
```
Maybe you can consider refactoring "deadline" into an enum, such that you can avoid hardcoding 8 over here, similar to the other commands

nhjryan replied:
Thanks for all the comments and suggestions (including the ones above)! Will keep these in mind! :)

cedricongjh reviewed (commented):
Overall, not much issues besides the minor nitpicks I commented on, well done!

meerian commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,101 @@
+                    System.out.println(" " + (i + 1) + "." + list.get(i));
+                }
+                System.out.println("____________________________________________________________");
+            } else if (input.equals("bye")) {
```
Perhaps you can do more OOP (such as having a Ui class) to avoid having a long method?

nhjryan replied:
Thanks for the suggestion! Have been bogged down by work lately, will definitely keep this in mind when refactoring my code! :)

meerian commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,22 @@
+todoo
+deadline
+deadline need to do homework
+bye
```
Good job with the rigorous testing using the text-ui-test!👍

nhjryan committed:
Add Level 7. Save

nhjryan committed:
Add Level 8. Dates and Times

nhjryan committed:
Add file to .gitignore

nhjryan committed:
Merge branch 'branch-Level-7'

nhjryan committed:
Merge and de-conflict branch 'branch-Level-8'

nhjryan committed:
Re-factor code for better clarity

nhjryan committed:
Divide classes into packages

nhjryan committed:
Add JUnit Tests for testing purposes

nhjryan committed:
Package the App as a JAR file

nhjryan committed:
Add JavaDoc comments

nhjryan committed:
Tweak the code to comply with a coding standard

nhjryan committed:
Give users a way to find a task by searching for a keyword.

nhjryan committed:
Merge branch 'branch-A-CodingStandard'

nhjryan committed:
Merge branch 'branch-Level-9'

nhjryan committed:
Implement GUI

nhjryan committed:
Add Gradle support

nhjryan committed:
Merge branch 'branch-Level-10'

nhjryan committed:
Add More OOP by re-factoring code and edited relevant test cases

nhjryan committed:
Organize into Packages and edit code to ensure accessibility

nhjryan committed:
Add JUnit Tests and make adjustments to parsing for Storage

nhjryan committed:
Create a JAR file and make necessary fixes for it

nhjryan committed:
Add Level 9. Find

nhjryan committed:
Add JavaDoc

nhjryan committed:
Follow the Coding Standard

nhjryan committed:
Merge branch 'branch-A-JavaDoc'

nhjryan committed:
Merge branch 'branch-A-CodingStandard'

nhjryan committed:
Merge remote-tracking branch 'origin/add-gradle-support'

nhjryan committed:
Fix issues with running Gradle

nhjryan committed:
Use CheckStyle and fix coding violations

nhjryan committed:
Add Level 10. GUI

nhjryan committed:
Add JavaDocs to the FindCommand and make edits to other Command classes

There are no JavaDocs for FindCommand and other Command classes are not
updated to reflect the return value changes.

JavaDocs need to be rectified to properly reflect the new updates due to
JavaFX.

Let's edit the JavaDocs to reflect the changes to the Duke program since
the last time JavaDocs were edited.

nhjryan committed:
Use Assertions

There are no assertions used in the code.

This makes it hard to identify errors caused by faulty methods.

Let's use assertions where applicable to assert conditions that should
hold true in the code.

nhjryan committed:
Improve Code Quality

The code quality does not meet the necessary standards.

This is not ideal for others to read.

Let's re-factor the code and make adjustments to ensure that the code
does not become overly convoluted.

nhjryan committed:
Merge pull request #2 from nhjryan/branch-A-Assertions

Use Assertions

nhjryan committed:
Merge branch 'master' into branch-A-CodeQuality

nhjryan committed:
Merge pull request #3 from nhjryan/branch-A-CodeQuality

Improve Code Quality

nhjryan committed:
Add Extension: C-MassOps

nhjryan committed:
Update README.md

nhjryan committed:
Set theme jekyll-theme-slate

nhjryan committed:
Set theme jekyll-theme-cayman

nhjryan committed:
Update README.md to match GitHub Pages format

nhjryan committed:
Update README.md again

nhjryan committed:
Update README.md for the third time

nhjryan committed:
Update README.md once again

nhjryan committed:
Add a User Guide and product screenshot.

nhjryan committed:
Merge branch 'master' of https://github.com/nhjryan/ip

nhjryan committed:
Make necessary edits for cross-platform use
